Author: Pino Toscano <pino@debian.org>
Description: preliminary GNU/Hurd support
 Adds "support" for GNU/Hurd by defining MAXPATHLEN and PATH_MAX where needed.
 Not forwarded, better fixes should be done upstream.
 Also disable the fatal warnings for the linker, as we might hit symbols
 which are not implemented and will always fail.
Forwarded: no
Last-Update: 2011-05-30
--- a/kdecore/kernel/kstandarddirs.cpp
+++ b/kdecore/kernel/kstandarddirs.cpp
@@ -73,6 +73,10 @@
 #include <QtCore/QFileInfo>
 #include <QtCore/QSettings>
 
+#ifndef MAXPATHLEN
+# define MAXPATHLEN 1024
+#endif
+
 class KStandardDirs::KStandardDirsPrivate
 {
 public:
--- a/kinit/lnusertemp.c
+++ b/kinit/lnusertemp.c
@@ -36,6 +36,10 @@
 #include <limits.h>
 #endif
 
+#ifndef PATH_MAX
+# define PATH_MAX 1024
+#endif
+
 int check_tmp_dir(const char *tmp_dir, int check_ownership);
 int create_link(const char *file, const char *tmp_dir);
 int build_link(const char* tmp, const char *tmp_prefix, const char *kde_prefix);
--- a/cmake/modules/FindKDE4Internal.cmake
+++ b/cmake/modules/FindKDE4Internal.cmake
@@ -1221,8 +1221,8 @@ if (CMAKE_COMPILER_IS_GNUCXX)
    endif (CMAKE_SYSTEM_NAME MATCHES Linux OR CMAKE_SYSTEM_NAME STREQUAL GNU)
 
    if (CMAKE_SYSTEM_NAME STREQUAL GNU)
-      set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -pthread")
-      set (CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -pthread")
+      set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -pthread -Wl,--no-fatal-warnings")
+      set (CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -pthread -Wl,--no-fatal-warnings")
    endif (CMAKE_SYSTEM_NAME STREQUAL GNU)
 
    # gcc under Windows
